lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -743,6 +743,45 @@ OpFoldResult MuxOp::fold(ArrayRef<Attribute> constants) {
   return {};
 }
 
+void MuxOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
+                                        MLIRContext *context) {
+  struct Folder final : public OpRewritePattern<MuxOp> {
+    using OpRewritePattern::OpRewritePattern;
+    LogicalResult matchAndRewrite(MuxOp op,
+                                  PatternRewriter &rewriter) const override {
+      auto width = op.getType().cast<IntegerType>().getWidth();
+
+      APInt value;
+
+      // mux(a, 11...1, b) -> or(a, b)
+      if (matchPattern(op.trueValue(), m_RConstant(value))) {
+        if (value.isAllOnesValue()) {
+          auto cond = width == 1 ? op.cond()
+                                 : rewriter.createOrFold<SExtOp>(
+                                       op.getLoc(), op.getType(), op.cond());
```

Comment:
Super nit, but I'd suggest writing this like this to make clang-format happier:

```
auto cond = op.cond();
if (width > 1)
  cond = rewriter.createOrFold<SExtOp>(op.getLoc(), op.getType(), cond);
```

here and below